### ISSUE TITLE
Custom css URI functions are breaking

### ISSUE THREAD

**Issue by vjeux** [prettier/prettier#4644]
**Prettier 1.13.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATAlgNwATADpTbYBGAhmANYDmAThAK5SpLYC0AZsa6qTKfTXQAKAPTpYcGlDEBbUlTgBnERBpURABzqp6YGCNRx2-ADYxW6OQoB0GqFQCUAbgJEylWgyYsOXHnwFhMQkpWXklFTVSKHQAL150aE1tXX1DY3ozCys4W3tnAgBfEAAaEAgNGESoRWRQUho6AHcABQaEWpRSTAh0VFKQYhpyCjgYAGUNcnEqZBgaejgy8UVJGBbhqjlkYxNVsoArRQAPACFhyjHx0hk4ABlxOB3SPaWQKZpVmmRwRVqyrTiGAAdT6MAAFsgABwABgBdFWwOGGh+WiUkkwTzKNDgAEd6OgcRt5NskLt9iBVjJ0HMFm9FDMTHAAIr0CDwZ6vMp8Yig1AQ5AAJm5w3QJhmAGEIDJSSAoNAsSB6KsACqkYidclwQqFIA)
```sh
--parser css
```

**Input:**
```css
div {
  background: -fb-datauri(/intern/images/org/product/default-image.png);
  background: -fb-datauri(/intern/images/organization/product/default-image.png);
}
```

**Output:**
```css
div {
  background: -fb-datauri(/ intern/images/org/product/default-image.png);
  background: -fb-datauri(
    / intern/images/organization/product/default-image.png
  );
}
```

This is not urgent but would be nice to find a way not to put a space after the first `/`. I remember that we are doing some custom things for urls, maybe we can whitelist everything that ends with url/uri as a heuristic?

**Comment by evilebottnawi** [comment]
@vjeux because `-fb-datauri` is considered as not `url` function. But we can handle this as `url`.

**Comment by vjeux** [comment]
Sounds good, thanks for the pointer!

**Comment by lydell** [comment]
@vjeux If it's possible to quote the URLs you can use that as a workaround.

**Prettier 1.13.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATAlgNwATADpTbYBGAhmANYDmAThAK5SpLYC0AZsa6qTKfTXQAKPCAD06WHBpQJAW1JU4AZzEQaVMQAc6qemBhjUcdvwA2MVugVKAdFqhVRASgDcBImUq0GTFhy4ePgFhUQkpGXlFFTUNUih0AC9edGhtXX1DY1N6CysbOHtHF3coAF8QABoQCC0YVKhlZFBSGjoAdwAFVoQmlFJMCHRUKpBiGnIKOBgAZS1ySSpkGBp6OGrJZWkYTomqBWRTMy3qgCtlAA8AIQnKaZnSOTgAGUk4Q9Jj9ZB5mi2aZDgZRNao6SQwADqwxgAAtkAAOAAMoLoWwhEy0gJ0KmkmHe1RocAAjvR0ITdooDkgjicQFs5OhlqtvspFmY4ABFegQeAfL7VPjEKGoWHIABMAom6DMiwAwhA5FSQFBoPiQPQtgAVUjEPo0uBlMpAA)
```sh
--parser css
```

**Input:**
```css
div {
  background: -fb-datauri("/intern/images/org/product/default-image.png");
  background: -fb-datauri("/intern/images/organization/product/default-image.png");
}
```

**Output:**
```css
div {
  background: -fb-datauri("/intern/images/org/product/default-image.png");
  background: -fb-datauri(
    "/intern/images/organization/product/default-image.png"
  );
}

```

**Comment by evilebottnawi** [comment]
Same problem https://github.com/prettier/prettier/issues/2835, try to fix tommorow

**Comment by vjeux** [comment]
I had some internal discussions and the current regex also doesn't support "" nor \n inside. Right now there's only a 100 occurrences so it's not a huge deal. There's a new infra to extract those pieces from CSS that's in the work that will support " but it's likely going to take some time to release.

I believe that we should find a way inside of prettier to not put a space after the / anyway, but in terms of priority it's pretty low and if we don't put any special handling inside of prettier we'll be fine, we can workaround on our end when it becomes an issue.

**Comment by dhruvdutt** [comment]
> can break many code using endWiths, better use list of this functions

@evilebottnawi Can you please explain this by some example?

**Comment by evilebottnawi** [comment]
@dhruvdutt 
```scss
.a {
  background: my-custom-scss-less-function('first_arg_is_not_url_but_can_have spaces and other strange-characters-+-*/');
}
```

**Comment by evilebottnawi** [comment]
@dhruvdutt 
1. Switch on `postcss-value-parse` (url function have own algorithm for parsing)
2. Use static array for `url` functions, but scss allow to redefine `url` function and it can break some edge cases like
```scss
@function url($arg) {
  return 'url'
}

a {
  background: url('same-other-strange-arg')
}
```
But some limitation for css is normal

**Comment by suchipi** [comment]
@vjeux still relevant? Should we pursue this or are you fine using prettier ignore comments and closing this issue?

**Comment by evilebottnawi** [comment]
@suchipi no, we should fix it, it is not easy, but possible

**Comment by suchipi** [comment]
I was gonna fix it really quick but didn't want to waste my time if it's not a problem for facebook anymore- vjeux indicated that they were changing the infra some around this

**Comment by vjeux** [comment]
We could put everything with " and then allow for spaces around. But if it was fixed within prettier that would be great. It's not urgent though, it only affects 40 files that are barely touched.

**Comment by Yajo** [comment]
I have [another failure case, using scss](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAlgWwA4QE4wAEArngDYAUA9FQGbQwDOAdAOYQStlwCG26LSJiphGjAPy0emdGQCeAXgDy2BAGoAyjyiMkAFgAMBgDSGD6GDzLowASgDcIYyAjYY6aI2SgeePBAB3AAVfBC8UKwCeOS9nACM8HjAAazgYDWwk9ChWZBg8YjhnAAsYTDIAdWKLOEZMsDgNMIt0ADcLOWRwMScQbMY4AiDE1kweZCkyAecAK0YADwAhRJS0rUw4ABlsuAmraZA5+Y1srjgARWIIeD2popBMvAG8LsZRWIe8bJgK9AATGDFZAADhMnwgAwqiWwXWweFqg1au2cAEcrvBhq5wiAeIwALRQOBwP7E3rwtHoeHDHijcZISYHAYyPIFe6MU7cS7XXb0-b3SxxX4AoFIABMznyPFkpwAwhBMGMurUAKy9YgDAAqPDi4QZ91ahQAklASbANGAvm4AIImjQwOTcW4DAC+zqAA):

```scss
@import url(//fonts.googleapis.com/css?family=Open+Sans:400,400italic);
```

is converted into:

```scss
@import url(//fonts.googleapis.com/css?family=Open + Sans:400, 400italic);
```

which is wrong syntax.

**Comment by fisker** [comment]
@Yajo Your case fixed in #7592

**Comment by thorn0** [comment]
Fixed by https://github.com/prettier/prettier/pull/9966
